python/ray/train/batch_predictor.py

```diff
@@ -213,7 +213,8 @@ def __call__(self, batch):
                 # Set the in-predictor preprocessing to a no-op when using a separate
                 # GPU stage. Otherwise, the preprocessing will be applied twice.
                 override_prep = BatchMapper(lambda x: x)
-                data = preprocessor.transform(data)
+                batch_fn = preprocessor._transform_batch
+                data = data.map_batches(batch_fn)
```

Comment:
Hmm do you know why the original code failed to work?

Reply:
Yeah, transform doesn't take pipelines, so it doesn't work.